File: backend/app-store/src/main/java/app/product/model/ProductDTO.java
package app.product.model;

import app.product.model.utill.ProductCategory;
import app.product.types.charger.model.ChargerDTO;
import app.product.types.cooler.model.CoolerDTO;
import app.product.types.dram_memory.model.DRAMMemoryDTO;
import app.product.types.graphic_card.model.GraphicCardDTO;
import app.product.types.hard_drive.model.HardDriveDTO;
import app.product.types.motherboard.model.MotherboardDTO;
import app.product.types.pc_case.model.PcCaseDTO;
import app.product.types.processor.model.ProcessorDTO;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class ProductDTO {
    private Long bid;
    private String thumbnail;
    private String pictures;
    private ProductCategory productCategory;
    private String name;
    private String producer;
    private String guarantee;
    private String model;
    private Double price;
    private int quantity;

    private ChargerDTO charger;
    private CoolerDTO cooler;
    private DRAMMemoryDTO dramMemory;
    private GraphicCardDTO graphicCard;
    private HardDriveDTO hardDrive;
    private MotherboardDTO motherboard;
    private PcCaseDTO pcCase;
    private ProcessorDTO processor;
}

File: frontend/app-store-ui/client/src/app/api/models/product-dto.ts
/* tslint:disable */
/* eslint-disable */
import { ChargerDto } from '../models/charger-dto';
import { CoolerDto } from '../models/cooler-dto';
import { DramMemoryDto } from '../models/dram-memory-dto';
import { GraphicCardDto } from '../models/graphic-card-dto';
import { HardDriveDto } from '../models/hard-drive-dto';
import { MotherboardDto } from '../models/motherboard-dto';
import { OpinionDto } from '../models/opinion-dto';
import { PcCaseDto } from '../models/pc-case-dto';
import { ProcessorDto } from '../models/processor-dto';
import { ProductCategory } from '../models/product-category';
export interface ProductDto {
  bid?: number;
  charger?: ChargerDto;
  cooler?: CoolerDto;
  dramMemory?: DramMemoryDto;
  graphicCard?: GraphicCardDto;
  guarantee?: string;
  hardDrive?: HardDriveDto;
  model?: string;
  motherboard?: MotherboardDto;
  name?: string;
  opinions?: Array<OpinionDto>;
  pcCase?: PcCaseDto;
  pictures?: string;
  price?: number;
  processor?: ProcessorDto;
  producer?: string;
  productCategory?: ProductCategory;
  quantity?: number;
  rate?: number;
  thumbnail?: string;
}
